### ISSUE TITLE
New rule: No chaining `this` to itself

### ISSUE THREAD

**Issue by roneesh** [ember-template-lint/ember-template-lint#2985]
While reviewing a PR at my work recently, we caught an instance of this being chained to this:

`this.this.foo`

And it didn't throw a lint error. I'm not seeing this in the docs either as a lint rule, so I propose that it be one (and happy to work on it!)

Here is an example app: https://github.com/roneesh/no-chained-this-example/blob/main/app/components/chained-this.hbs

**Comment by roneesh** [comment]
Here's a screencap of that component with the server running
<img width="707" alt="example of browser rendering of linked component" src="https://github.com/ember-template-lint/ember-template-lint/assets/2407832/c9e324f1-98e1-4132-902d-f98c8701f841">

**Comment by lifeart** [comment]
Ai generated rule description:

#no-chained-this

✅ The `extends: 'recommended'` property in a configuration file enables this rule.

This rule disallows chaining `this.this` inside templates — such as `{{this.this.foo}}` or `<this.this.bar />`. These patterns are misleading, usually unintentional, and result in unnecessary ambiguity about scope and component context.

## Motivation

Templates are meant to clearly reference local properties, arguments (`@arg`), or component state via `this`. When you see `this.this.foo`, it's either:

- A typo (e.g., copy/paste or incorrect refactor)
- A misunderstanding of Glimmer component boundaries
- A misuse of dynamic component invocation (e.g., `<this.this.foo />`)

These patterns often go unnoticed but produce confusing or broken runtime behavior.

## Examples

This rule **forbids** the following:

```hbs
{{this.this.value}}
```

```hbs
{{helper value=this.this.foo}}
```

```hbs
<this.this.Component />
```

```hbs
{{#if this.this.condition}}
  ...
{{/if}}
```

```hbs
{{component this.this.dynamicComponent}}
```

This rule **allows** the following:

```hbs
{{this.value}}
```

```hbs
<this.Component />
```

```hbs
{{component this.dynamicComponent}}
```

```hbs
{{@argName}}
```

## Suggested Error Message

> 🚫 `this.this.*` is not allowed in templates. This is likely a mistake — remove the redundant `this.`

## Migration

Just remove the extra `this`:

Before:

```hbs
{{this.this.foo}}
<this.this.bar />
```

After:

```hbs
{{this.foo}}
<this.bar />
```

## Configuration

- `boolean` — `true` to enable / `false` to disable

## References

- [Ember Guides – Glimmer Component Templates](https://guides.emberjs.com/release/components/glimmer-components/)
- [RFC: Strict Mode Template Context](https://github.com/emberjs/rfcs/blob/master/text/0496-template-strict-mode.md)

**Comment by lifeart** [comment]
@roneesh thank you for rule idea!
